python/tidy/servo_tidy/tidy.py

```diff
@@ -723,13 +725,32 @@ def parse_config(content):
     config['ignore']['directories'] = map(lambda path: os.path.join(*path.split('/')),
                                           config['ignore']['directories'])
 
+    # Add dict of dir, list of expected ext to config
+    dirs_for_ext = config_file.get("check_dirs_for_exts", {})
```

Comment:
I also have another small problem :)

Currently I pass the directories to check as an argument to the `check_directory_files` function, inside the main `scan` function, by actually grabbing from the config([see](https://github.com/gterzian/servo/blob/check_webidls_folder_files/python/tidy/servo_tidy/tidy.py#L855)). This makes testing that function very easy, as in the test I just pass another list of directories ([see](https://github.com/gterzian/servo/blob/check_webidls_folder_files/python/tidy/servo_tidy_tests/test_tidy.py#L39))

The funny thing is that, at this point in the scan function, the config hasn't been parsed yet. The parsing only occurs later, during the first iteration of the generators :) 

So if I remove the default key on top, I get a KeyError at that point.

 If I keep the default key, it works, because the initially empty dict is later mutated during parsing, so by the time the exectution gets to [here](https://github.com/gterzian/servo/blob/check_webidls_folder_files/python/tidy/servo_tidy/tidy.py#L743) we have the value we expect. 

So the question is, should I move the access to `config` to inside `check_directory_files`? If I do, it makes for harder testing of that function, since I would have to mock the config object.